framework/src/modules/reward/module.ts

```diff
@@ -99,12 +99,14 @@ export class RewardModule extends BaseModule {
 			throw new Error("Block reward can't be negative.");
 		}
 
-		await this._tokenAPI.mint(
-			context.getAPIContext(),
-			context.header.generatorAddress,
-			this._tokenID,
-			blockReward,
-		);
+		if (blockReward !== BigInt(0)) {
+			await this._tokenAPI.mint(
+				context.getAPIContext(),
+				context.header.generatorAddress,
+				this._tokenID,
+				blockReward,
+			);
+		}
```

Comment:
I think if the reward is not minted, event also doesn't make much sense?

Reply:
hmm I think event data will never contain, for ex, `REWARD_REDUCTION_SEED_REVEAL` as the reward can be zero and will never emit an event?